src/main/kotlin/pl/edu/uj/ii/ksi/mordor/services/FileEntryService.kt

```diff
@@ -0,0 +1,27 @@
+package pl.edu.uj.ii.ksi.mordor.services
+
+import java.io.File
+import org.springframework.stereotype.Service
+import pl.edu.uj.ii.ksi.mordor.persistence.entities.FileEntry
+import pl.edu.uj.ii.ksi.mordor.persistence.repositories.FileEntryRepository
+import pl.edu.uj.ii.ksi.mordor.services.repository.RepositoryService
+
+@Service
+class FileEntryService(
+    private val repositoryService: RepositoryService,
+    private val entryRepository: FileEntryRepository
+) {
+    fun fileExists(entry: FileEntry): Boolean {
+        return repositoryService.getEntity(entry.path) != null
```

Comment:
implement this in repositoryService to avoid building a full RepositoryEntity on each call.